app/services/migration/failures_manager.rb

```diff
@@ -0,0 +1,60 @@
+module Migration
+  class FailuresManager
+    class << self
+      def record_failures(data_migration:, items:)
+        new(data_migration:, items:).record_failures
+      end
+
+      def failures_for(data_migration:)
+        new(data_migration:).failures
```

Comment:
I think having the class and instance methods makes this interface more confusing than it needs to be; can we settle on one or the other? personally I think having instance methods makes it all neater, then I think we only really need two methods, one to record a failure and one to get all recorded failures:

```
failure_manager = FailureManager.new(data_migration:)
failure_manager.record_failure(item, failure_message)
failure_manager.all_failures
```